### DISCUSSION TITLE
How do you separate dependency of frontend and backend in an Nx workspace?

### DISCUSSION THREAD

**Discussion by winston0410** [nrwl/nx-console#1286] [Q&A]
So I want to share dependencies between packages in frontend and backend, but not having all of them in root. Otherwise when I run the CI when the backend's code has changed, I will have to download the dependencies for frontend as well. Can Nx handle that?

**Comment by Cammisuli** [comment]
Nx strongly recommends using a single version dependency. That means that your root package.json file should have the dependencies for all your apps/libs.

**Reply by Cammisuli** [reply]
You can also use a yarn/npm/pnpm/lerna workspace to handle your dependencies instead. Then add Nx to run the tasks: https://nx.dev/migration/adding-to-monorepo

**Reply by kasir-barati** [reply]
@Cammisuli can you elaborate what you mean by "strongly recommends"? The way I look at it, it's gonna unnecessarily increase the `node_modules` size installed, unwanted CI/CD pipelines execution, and polluting dependencies.

But I can also empathise what it means to have only one `package.json` and `package-lock.json`. It is much much easier to manage and operate with just one rather than having multiple one.

But nonetheless I still like to hear your comments on this.
